apps/hip/models.py

```diff
@@ -184,6 +274,75 @@ def get_context(self, request):
         return context
 
 
+class ListRowBlock(blocks.StructBlock):
+    page = blocks.PageChooserBlock(
+        required=True,
+        help_text=("An internal page"),
+    )
+    description = blocks.RichTextBlock(
+        required=False,
+        help_text=("Description for this row"),
+    )
+
+
+class ListRowStreamBlock(blocks.StreamBlock):
+    rows = ListRowBlock()
+
+
+class ListSectionBlock(blocks.StructBlock):
+    header = blocks.CharBlock(
+        max_length=80,
+        required=False,
+        help_text=("The heading for this section of rows (maximum of 80 characters)."),
+    )
+    show_header_in_right_nav = blocks.BooleanBlock(
+        required=False,
+        default=True,
+        help_text="Should this header be shown in the navigation on the right side of the page?",
+    )
+    rows = ListRowStreamBlock()
+
+
+class ListPage(HipBasePage):
+    show_breadcrumb = models.BooleanField(
```

Comment:
I guess I'm still a bit foggy on where the `ListPage` is to be made available and when this page is a child of another list page or specifically Homepage for that matter, how are are these ad hoc pages to be represented within the parent page's template. For example if a ListPage is a child of the Homepage, then should the ListPage show up in the sidebar for navigation?

Reply:
I'm wondering if a larger team discussion is needed to answer this? Overall, I really like your implementation!

Reply:
Cool, thanks!
I think the idea is that the `ListPage` can be available at any level of the hierarchy, and it's up to the CMS editors to make things look the way they want. The left navbar is not currently editable by CMS users, and will continue to not be editable by them. I'm happy to have a team discussion about it as well.
For example, CMS users may want to choose what shows up in the left navbar, and I think Wagtail already offers fields for that, but this project disables them. Maybe those fields are a better way to make it clear what shows up in the left navbar?
Realistically, though, we'll probably run out of time before making it work entirely intuitively